core-web/libs/sdk/client/src/lib/client/content/builders/collection/collection.ts

```diff
@@ -0,0 +1,334 @@
+
+        return this;
+    }
+
+    /**
+     * Takes an array of constrains to sort the content by field an specific order
+     *
+     * @example
+     * ```javascript
+     * // This will sort the content by title in ascending order
+     * // and by modDate in descending order
+     *  const sortBy = [{ field: 'title', order: 'asc' }, { field: 'modDate', order: 'desc' }]
+     *
+     *  client.content.getCollection("Blog").sortBy(sortBy)
+     *```
+     *
+     * @param {SortBy[]} sortBy Array of constrains to sort the content by
+     * @return {CollectionBuilder} CollectionBuilder - A CollectionBuilder instance
+     * @memberof CollectionBuilder
+     */
+    sortBy(sortBy: SortBy[]): this {
+        this.#sortBy = sortBy;
+
+        return this;
+    }
+
+    /**
+     * Takes a number that represents the max amount of content to fetch
+     *
+     * Note: The limit is set to 10 by default
+     *
+     * @param {number} limit The max amount of content to fetch
+     * @return {CollectionBuilder} CollectionBuilder - A CollectionBuilder instance
+     * @memberof CollectionBuilder
+     */
+    limit(limit: number): this {
+        this.#limit = limit;
+
+        return this;
+    }
+
+    /**
+     * Takes a number that represents the page to fetch
+     *
+     * @param {number} page The page to fetch
+     * @return {CollectionBuilder} CollectionBuilder - A CollectionBuilder instance
+     * @memberof CollectionBuilder
+     */
+    page(page: number): this {
+        this.#page = page;
+
+        return this;
+    }
+
+    /**
+     * Takes a string that represents a Lucene Query that is used to filter the content to fetch.
+     *
+     * Note: The string is not validated, so be cautious when using it.
+     *
+     * @param {string} query A Lucene Query String
+     * @return {CollectionBuilder} CollectionBuilder - A CollectionBuilder instance
+     * @memberof CollectionBuilder
+     */
+    query(query: string): this;
+
+    /**
+     * Takes a function that recieves a QueryBuilder to buid a query for content filtering.
+     * @example
+     *```javascript
+     * // This will filter the content by title equals 'Hello World' or 'Hello World 2'
+     * client.content.getCollection("Activity").query((queryBuilder) =>
+     *     queryBuilder.field('title').equals('Hello World').or().equals('Hello World 2')
+     * );
+     *```
+     * @param {BuildQuery} buildQuery A function that receives a QueryBuilder instance and returns a valid query
+     * @return {CollectionBuilder} CollectionBuilder - A CollectionBuilder instance
+     * @memberof CollectionBuilder
+     */
+    query(buildQuery: BuildQuery): this;
+    query(buildQuery: BuildQuery | string): this {
+        if (typeof buildQuery === 'string') {
+            this.#rawQuery = buildQuery;
+
+            return this;
+        }
+
+        if (typeof buildQuery !== 'function') {
+            throw new Error('Parameter for query method should be a function or a string');
```

Comment:
```suggestion
            throw new Error('Parameter for query method should be a ???? or a string');
```

Not a function a query something no?

Reply:
The error should say how I fix it...